toolchain/diagnostics/diagnostic_emitter.h

```diff
@@ -367,7 +371,17 @@ class StreamDiagnosticConsumer : public DiagnosticConsumer {
     if (message.location.column_number > 0) {
       *stream_ << message.location.line << "\n";
       stream_->indent(message.location.column_number - 1);
-      *stream_ << "^\n";
+      *stream_ << "^";
+      // We want to confirm the underlined portion is atleast 1 '~' long but it
+      // does not go past the end of the line in the case of a multiline token.
+      int underline_length = std::max(1, message.location.length - 1);
+      underline_length = std::min(
```

Comment:
Note the diagnostics such as:

```
// CHECK:STDERR: var a: [1; 39999999999999999993];
// CHECK:STDERR:                                ^~
```

Here, only the close `]` should be highlighted. The `~` indicates the `;` is also part of the error, which seems incorrect; @zygoloid was this your intent?

Reply:
No, sorry, I meant max(0, ...) in the second option!

Reply:
Ah, ok, I thought that was strange. Just updated it.